feat: add appium 2.0 migration guide (#1568)
saucelabs/sauce-docs: docs/mobile-apps/automated-testing/appium/appium-2-migration.md

---
id: appium-2-migration
title: Migrating to Appium 2.0 on Sauce Labs
sidebar_label: Appium 2.0 Migration
description: Learn how to migrate you Appium 1 tests to Appium 2 on Sauce Labs.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

:::warning Appium 1 End-of-life

> The Appium core team does not maintain Appium 1.x anymore since the 1st of January 2022. All recent versions of officially supported platform drivers are not compatible to Appium 1.x anymore, and require Appium 2 to run.

Reviewer: ```suggestion
The Appium core team does not maintain Appium 1.x anymore since the 1st of January 2022. Recent versions of all officially supported platform drivers are no longer compatible with  Appium 1.x, and require Appium 2 to run.
```

:::

> Appium 2.0 is the most major new release of Appium in the last years. The changes in Appium 2.0 are not primarily related to changes in automation behaviors for specific platforms. Instead, Appium 2.0 re-envisions Appium as a platform where "drivers" (code projects that introduce support for automation of a given platform) and "plugins" (code projects that allow for overriding, altering, extending, or adding behaviors to Appium) can be easily created and shared.

Reviewer: ```suggestion
The changes in Appium 2.0 are not primarily related to changes in automation behaviors for specific platforms. Instead, Appium 2.0 re-envisions Appium as a platform where "drivers" (code projects that introduce support for automation of a given platform) and "plugins" (code projects that allow for overriding, altering, extending, or adding behaviors to Appium) can be easily created and shared.
```

At the same time, the Appium project is taking the opportunity to remove many old and deprecated bits of functionality. More information about this can be found in the [Appium 2.0 release notes](https://github.com/appium/appium/releases).

:::tip Appium 2.0 Documentation
Appium is in the final stages of a major revision (to version 2.0). As such, the documentation found around the web may not be correct. The current Appium 2.0 documentation is very much in progress. Currently, it can be found [here](https://appium.github.io/appium/docs/en/2.0/).
:::

## Breaking Changes

> Before you get started with migrating your Appium 1 tests to Appium 2, you should have a look a the [Appium 2.0 release notes](https://github.com/appium/appium/releases). The breaking changes that are listed below are only a subset of the changes that are listed in the release notes, but also have an impact on running your tests on Sauce Labs.

Reviewer: ```suggestion
Before you get started with migrating your Appium 1 tests to Appium 2, read the [Appium 2.0 release notes](https://github.com/appium/appium/releases). The breaking changes that are listed below are only a subset of the changes that are listed in the release notes, but also have an impact on running your tests on Sauce Labs.
```

## Getting started with Appium 2.0 on Sauce Labs

:::caution Appium 2.0 progress on Sauce Labs
Sauce Labs is currently in the process of implementing Appium 2.0 into our Virtual and Real Device cloud. We are working hard to make Appium 2.0 available for all our customers as soon as possible. Please check back here for updates.

| Cloud           | Status                                                                                     |
| --------------- | ------------------------------------------------------------------------------------------ |
| Real Devices    | Available, see [supported drivers](#supported-drivers) for all available drivers           |
| Virtual Devices | Partially available, see [supported drivers](#supported-drivers) for all available drivers |

:::

As mentioned above, the following breaking changes lead to using the following capabilities:

- [W3C capabilities](#appium-capabilities)
- [Required `automationName`](#automationname-is-now-required)
- [Sauce Labs specific capabilities](#sauce-labs-specific-capabilities)

### Capabilities

> Below you can find an example of a capabilities object for running a test on Sauce Labs with Appium 2.0. The examples are created for Real Devices, for Virtual Devices we advise you to use our [Platform Configurator](https://saucelabs.com/platform/platform-configurator#/).

Reviewer: ```suggestion
Below you can find an example of a capabilities object for running a test on Sauce Labs with Appium 2.0. The examples are created for Real Devices, for Virtual Devices you can use our [Platform Configurator](https://saucelabs.com/platform/platform-configurator#/).
```

> Below you can find an example of a capabilities object for running a test on Sauce Labs with Appium 2.0. The examples are created for Real Devices, for Virtual device we advice you to check out our [Platform Configurator](https://saucelabs.com/platform/platform-configurator#/).

Reviewer: ```suggestion
Below you can find an example of a capabilities object for running a test on Sauce Labs with Appium 2.0. The examples are created for Real Devices, for Virtual Devices we advise you to use our [Platform Configurator](https://saucelabs.com/platform/platform-configurator#/).
```

---
id: appium-2-migration
title: Migrating to Appium 2.0 on Sauce Labs

> sidebar_label: Appium 2 Migration

Reviewer 2: ```suggestion
sidebar_label: Appium 2.0 Migration
```
